src/org/openstreetmap/josm/actions/search/SearchCompiler.java

```diff
@@ -1554,6 +1558,44 @@ public String toString() {
         }
     }
 
+    /**
+     * Matches presets.
+     */
+    private static class Preset extends Match {
+        private List<TaggingPreset> ps;
+
+        Preset(String presetName) throws ParseError {
+
+            if (presetName == null)
+                throw new ParseError("Preset name cannot be null");
+
+            Collection<TaggingPreset> ts = TaggingPresets.getTaggingPresets();
+            ps = new ArrayList<>();
+
+            for (TaggingPreset t : ts) {
+                String name = t.getSimpleName();
+
+                if (name != null && name.equalsIgnoreCase(presetName)) ps.add(t);
+            }
+
+            if (ps.isEmpty())
+                throw new ParseError(tr("Unknown preset name: ") + presetName);
+        }
+
+        /**
+         * Since presets can have common names, the primitive is considered to
+         * belong to a certain preset if it matches at least one of them.
+         */
+        @Override
+        public boolean match(OsmPrimitive osm) {
```

Comment:
(just a short note: Converting this to streams would not be good - streams don't have a good performance for this)